dbcon/mysql/ha_mcs_pushdown.cpp

```diff
@@ -892,10 +888,26 @@ int ha_columnstore_select_handler::init_scan()
 {
     DBUG_ENTER("ha_columnstore_select_handler::init_scan");
 
-    // Dummy init for SH. Actual init happens in create_SH
-    // to allow fallback to other handlers if SH fails.
     int rc = 0;
 
+    if (!rewrite_error)
+    {
+        // handler::table is the place for the result set
+        // Skip execution for EXPLAIN queries
+        if (!thd->lex->describe)
```

Comment:
Curious, why was this moved from create_SH to here?

Reply:
Good question.
B/c create_SH is inappropriate place to execute the query(we can't even return and error w/o breaking the execution and init_SH is the place to execute the query.
I placed execution in create_SH in the first place to have a fallback but later understood it could break execution when we partially rewrote SELECT_LEX and then fallback to table API